micromanager_gui/main_window.py

```diff
@@ -152,19 +175,126 @@ def add_frame_explorer(self, name, array):
         except KeyError:
             self.viewer.add_image(array, name=layer_name)
 
-    # TO DO: add the file name form the save box
+
+    def _on_mda_started(self, sequence: useq.MDASequence):
+        """"create temp folder when mda starts."""
+
+        self.temp_folder = tempfile.TemporaryDirectory(None, str(sequence.uid))
+
+        """blok gui when mda starts."""
+        self.mda.disable_mda_groupbox()
+        self.disable_gui()
+
+    
+    def get_filename(self, fname, list_dir):
```

Comment:
I see now that you've gathered `list_dir` to check for the existence of files...   pathlib already has an [`exists`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.exists) function for this... so no need to hold on to `list_dir` just for that.  Just build the string you would like to check, and then use either `os.path.exists('some_name')` or use `Path('some_name').exists()` ... don't check to see if it is in the list